ha/core/system_health/system_health.py

```diff
@@ -0,0 +1,191 @@
+            component = SystemHealthComponents.get_component(healthevent.resource_type)
+
+            # Get the health update hierarchy
+            self.update_hierarchy = SystemHealthHierarchy.get_hierarchy(component)
+            if (len(self.update_hierarchy) - 1) > self.update_hierarchy.index(component):
+                next_component = self.update_hierarchy[self.update_hierarchy.index(component) + 1]
+            else:
+                next_component = None
+
+            # Get the component type and id received in the event.
+            component_type = healthevent.resource_type.split(':')[-1]
+            component_id = healthevent.resource_id
+            # Read the currently stored health value
+            current_health = self.get_status(component, component_id, comp_type=component_type,
+                                        cluster_id=healthevent.cluster_id, site_id=healthevent.site_id,
+                                        rack_id=healthevent.rack_id, storageset_id=healthevent.storageset_id,
+                                        node_id=healthevent.node_id, server_id=healthevent.node_id,
+                                        storage_id=healthevent.node_id)
+            if current_health:
+                # Update the current health value itself.
+                updated_health = EntityHealth.read(current_health)
+            else:
+                # Health value not present in the store currently, create now.
+                updated_health = EntityHealth()
+
+            # Create a new event and action
+            current_timestamp = str(int(time.time()))
+            entity_event = EntityEvent(healthevent.timestamp, current_timestamp, status, healthevent.specific_info)
+            entity_action = EntityAction(current_timestamp, const.ACTION_STATUS.PENDING.value)
+            # Add the new event and action to the health value
+            updated_health.add_event(entity_event)
+            updated_health.set_action(entity_action)
+            # Convert the health value as appropriate for writing to the store.
+            updated_health = EntityHealth.write(updated_health)
+
+            # Update the node map
+            self.node_id = healthevent.node_id
+            self.node_map = {'cluster_id':healthevent.cluster_id, 'site_id':healthevent.site_id,
+                             'rack_id':healthevent.rack_id, 'storageset_id':healthevent.storageset_id}
+
+            # Update in the store.
+            self._update(component, component_type, component_id, updated_health, next_component=next_component)
+            Log.info(f"Updated health for component: {component}, Type: {component_type}, Id: {component_id}")
+
+        except Exception as e:
+            Log.error(f"Failed processing system health event with Error: {e}")
+            raise HaSystemHealthException("Failed processing system health event")
+
+def main(argv: dict):
+    pass
+
+if __name__ == '__main__':
```

Comment:
`if __name__ == '__main__':`
is it for testing purpose

can we move it in test/integration/test_system_health.py

also 
`sys.path.append(os.path.join(os.path.dirname(pathlib.Path(__file__)), '..', '..', '..'))`
moved to test

Reply:
Sure, we could add test cases to this file later.

Reply:
done